### DISCUSSION TITLE
Aplicativos Web

### DISCUSSION THREAD

**Discussion by lclaros85** [SCADA-LTS/Scada-LTS#3065] [Ideas]
rrently, I work at a public company that uses ScadaBR 1.0 + MySQL and PHPMyAdmin, with .jsp apps. I migrated to version 1.2 and it worked fine, but now I need to move to LTS, and that's where it gets complicated. How can I make the LTS version use JSP and bypass the login to go directly to the application? In version 1.2, it works normally.

**Comment by fabiodurao** [comment]
Hello @lclaros85, ideally the conversation here should be in English, but since I know your needs, I will try to explain things in more detail so that Kamil or another colleague can help us.

In ScadaBR, JSP files were often used inside the folder to allow other third-party apps to consume or write data to ScadaBR in the same way that an API would.

A simple reading example using JSP would be something like

<%@ include file="/WEB-INF/jsp/include/tech.jsp" %>
<%@ taglib prefix="view" tagdir="/WEB-INF/tags/view" %>
<link href="resources/common.css" type="text/css rel="stylesheet"/>
<view:init username="admin"/>
<view:staticPoint xid="DP_014693"/>

I personally haven't tested this type of solution on Scada-LTS yet, but I know that @lclaros85 is trying to migrate something ready from ScadaBR to Scada-LTS without success.

I suggested using the REST API, but to have the least impact on the migration he would like to keep what is already ready. I imagine that the standard format may need an additional layer of authentication, due to the various security improvements that Scada-LTS has. received, could you help us with this @Limraj ?

**Reply by lclaros85** [reply]
thank you

**Comment by Limraj** [comment]
Hi @lclaros85 and @fabiodurao,
why can't you login? Access to application data by login in is a rather natural approach in web applications...

When it comes to view access for external users, we suggest the following approach:

**Good solution:**
1. Login as admin;
2. Create a profile pro_1 (in 'Users profile' tab) with permissions to the view and all points in the view:
![image](https://github.com/user-attachments/assets/a5c18d8c-ef2c-485b-b923-95e2568fbed4)
![image](https://github.com/user-attachments/assets/01660c08-701d-41b3-8f9b-fee8352279f2)

3. Create an account for the user by setting the profile pro_1:
![image](https://github.com/user-attachments/assets/b0f96196-f4a2-40b7-92f5-f875043b55b9)
The Home URL causes the user, after login, to be redirect immediately to the Home page here with views;

4. Then go to 'System settings' and select the following options:
![image](https://github.com/user-attachments/assets/9fb18d6d-2dd5-475e-a9b4-a049082fb4a3)
Thanks to this, the user will open full-screen mode with hidden information on how to exit this mode;

5. Now user_1, after login, will be redirected to the views page, with forced fullscreen mode:
![image](https://github.com/user-attachments/assets/c8184bf7-1fc6-4fd8-9d96-92915af03362)
Only sees the shared view.

6. If login on other user then go to in browser:
http://127.0.0.1:8080/Scada-LTS/logout.htm
Then show login page;

End

**Not recommended solution:**
We provide a public view function, to enable it you need to:
1. Log in as admin;
2. Edit the view that we want to make publicly available, change the permissions to read if we are only interested in reading values ​​in the view or set if we want to set values ​​from this public view:
![image](https://github.com/user-attachments/assets/d7d5e7fe-0b6d-4e3d-96dc-e1e2d74a8b01)
3. Next click on 'Users' and enabled user 'anonymous-user' (uncheck disabled checkbox);
![image](https://github.com/user-attachments/assets/7a57302b-5c24-4728-b9e7-f55f28eab1c9)
5. It would be best to configure a special "User profile" for this user in the "Users Profile" tab to set the appropriate permissions for the view that we want to make publicly available, as well as the Data Source and Data Point that are in this view:
![image](https://github.com/user-attachments/assets/a5c18d8c-ef2c-485b-b923-95e2568fbed4)
![image](https://github.com/user-attachments/assets/01660c08-701d-41b3-8f9b-fee8352279f2)

And set profile 'pro_1' in 'anonymous-user' -> Click on 'Users', select 'anonymous-user', and select profile 'pro_1' and 'save':
![image](https://github.com/user-attachments/assets/a6c242fc-a1a2-413d-89e8-f4a9b1c77b99)

Or similarly, it is very dangerous to set 'anonymous-user' as admin, click on 'Users' select 'anonymous-user' and check 'administrator' checkbox, and save;
![image](https://github.com/user-attachments/assets/40436e58-697f-48f7-8841-c971ba3e4e15)

8. Then logout and open:
http://127.0.0.1:8080/Scada-LTS/public_view.htm?viewId=viewId_to_public

**Unfortunately, we have limited resources and cannot deal with everything at once. However, we encourage commercial cooperation. You would be in direct contact with us and your problems would be a priority.**

Regards.

**Comment by Limraj** [comment]
Hi @fabiodurao Can you be more clear? What does this laugh mean? What happened?

**Reply by fabiodurao** [reply]
Hello Kamil, I have been following @lclaros85 project, they are from a municipal sanitation company, my emoji was just indicating that I was happy with the support you gave us. Thank you.

**Reply by lclaros85** [reply]
I also really appreciate the support, I'm not yet experienced in scadaLTS, who knows, I can improve with help, thanks again.

**Reply by Limraj** [reply]
@fabiodurao I understand, but it's not a smiling emoticon, this emoticon more means something like 'ha ha', so I guess it's not very suitable for showing that you like something, unless it's funny. It's called "laugh" not smile. In communication, it is important that our reactions are clear.
